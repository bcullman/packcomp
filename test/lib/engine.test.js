const fetch = require('node-fetch');

jest.mock('node-fetch');

const {
  getContents,
  getTitles,
  getFilter,
  getDependencies,
} = require('../../lib/engine');

const packageFixture3 = require('../fixtures/3/package.json');

describe('engine', () => {
  beforeEach(() => {
    fetch.mockReset();
  });

  test('should get contents', async () => {
    const argv = {
      _: [
        'test/fixtures/1/',
        'test/fixtures/2/package.json',
        'test/fixtures/3/',
      ],
    };
    const contents = await getContents(argv);
    expect(contents).toMatchSnapshot();
  });

  test('should get titles', () => {
    const argv = {
      _: [
        'test/fixtures/1/',
        'test/fixtures/2/package.json',
        'test/fixtures/3/',
      ],
    };
    const contents = getTitles(argv);
    expect(contents).toMatchSnapshot();
  });

  test('should get dependencies', async () => {
    const argv = {
      _: [
        'test/fixtures/1/',
        'test/fixtures/2/package.json',
        'test/fixtures/3/',
      ],
    };
    const contents = await getDependencies(argv);
    expect(contents).toMatchSnapshot();
  });

  test('should get dependencies with URLs', async () => {
    const argv = {
      _: [
        'test/fixtures/1/',
        'test/fixtures/2/package.json',
        'https://some-domain.com/some-package.json',
      ],
    };

    const resp = { status: 200, json: () => Promise.resolve(packageFixture3) };
    fetch.mockResolvedValue(resp);

    const contents = await getDependencies(argv);
    expect(contents).toMatchSnapshot();
  });

  test('should work when dependencies are identical', async () => {
    const argv = {
      _: [
        'test/fixtures/1/',
        'test/fixtures/1/package.json',
        'test/fixtures/1/',
      ],
    };
    const contents = await getDependencies(argv);
    expect(contents).toMatchSnapshot();
  });

  test('should get * for no filter', () => {
    const filter = getFilter({});
    expect(filter).toBe('*');
  });

  test('should get string for filter', () => {
    const filter = getFilter({ f: 'xyz' });
    expect(filter).toBe('xyz');
  });
});
