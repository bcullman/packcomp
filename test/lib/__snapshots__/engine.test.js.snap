// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engine should get contents 1`] = `
Array [
  Object {
    "dependencies": Object {
      "moment": "2.19.4",
    },
    "devDependencies": Object {
      "babel-cli": "6.26.0",
      "babel-core": "6.26.0",
      "babel-eslint": "8.0.3",
      "babel-preset-es2015": "6.24.1",
      "babel-preset-stage-1": "6.24.1",
      "babel-register": "6.26.0",
      "eslint": "4.13.1",
      "eslint-config-airbnb-base": "12.1.0",
      "eslint-plugin-security": "1.4.0",
      "mocha": "4.0.1",
      "nyc": "11.3.0",
      "pre-commit": "1.2.2",
      "rimraf": "2.6.2",
    },
  },
  Object {
    "dependencies": Object {
      "async": "2.6.1",
      "debug": "3.1.0",
      "lodash": "4.17.10",
      "nodemailer": "4.6.5",
      "request": "2.87.0",
      "string": "3.3.3",
      "sugar": "2.0.4",
    },
    "devDependencies": Object {
      "bunyan": "1.8.12",
      "chai": "4.1.2",
      "coveralls": "3.0.1",
      "eslint": "4.19.1",
      "eslint-config-guyellis": "0.0.6",
      "istanbul": "0.4.5",
      "mocha": "5.2.0",
      "npm-update-outdated": "0.1.6",
      "pre-commit": "1.2.2",
      "rewire": "4.0.1",
    },
  },
  Object {
    "dependencies": Object {
      "console.table": "0.10.0",
      "debug": "3.1.0",
      "lodash": "4.17.10",
      "minimist": "1.2.0",
    },
    "devDependencies": Object {
      "eslint": "4.19.1",
      "eslint-config-airbnb-base": "12.1.0",
      "eslint-plugin-import": "2.12.0",
      "eslint-plugin-jest": "21.17.0",
      "jest": "23.1.0",
    },
  },
]
`;

exports[`engine should get dependencies 1`] = `
Object {
  "dependencies": Array [
    Object {
      "1": "2.19.4",
      "2": "missing",
      "3": "missing",
      "Module": "moment",
    },
    Object {
      "1": "missing",
      "2": "2.6.1",
      "3": "missing",
      "Module": "async",
    },
    Object {
      "1": "missing",
      "2": "3.1.0",
      "3": "3.1.0",
      "Module": "debug",
    },
    Object {
      "1": "missing",
      "2": "4.17.10",
      "3": "4.17.10",
      "Module": "lodash",
    },
    Object {
      "1": "missing",
      "2": "4.6.5",
      "3": "missing",
      "Module": "nodemailer",
    },
    Object {
      "1": "missing",
      "2": "2.87.0",
      "3": "missing",
      "Module": "request",
    },
    Object {
      "1": "missing",
      "2": "3.3.3",
      "3": "missing",
      "Module": "string",
    },
    Object {
      "1": "missing",
      "2": "2.0.4",
      "3": "missing",
      "Module": "sugar",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "3": "0.10.0",
      "Module": "console.table",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "3": "1.2.0",
      "Module": "minimist",
    },
  ],
  "devDependencies": Array [
    Object {
      "1": "6.26.0",
      "2": "missing",
      "3": "missing",
      "Module": "babel-cli",
    },
    Object {
      "1": "6.26.0",
      "2": "missing",
      "3": "missing",
      "Module": "babel-core",
    },
    Object {
      "1": "8.0.3",
      "2": "missing",
      "3": "missing",
      "Module": "babel-eslint",
    },
    Object {
      "1": "6.24.1",
      "2": "missing",
      "3": "missing",
      "Module": "babel-preset-es2015",
    },
    Object {
      "1": "6.24.1",
      "2": "missing",
      "3": "missing",
      "Module": "babel-preset-stage-1",
    },
    Object {
      "1": "6.26.0",
      "2": "missing",
      "3": "missing",
      "Module": "babel-register",
    },
    Object {
      "1": "4.13.1",
      "2": "4.19.1",
      "3": "4.19.1",
      "Module": "eslint",
    },
    Object {
      "1": "12.1.0",
      "2": "missing",
      "3": "12.1.0",
      "Module": "eslint-config-airbnb-base",
    },
    Object {
      "1": "1.4.0",
      "2": "missing",
      "3": "missing",
      "Module": "eslint-plugin-security",
    },
    Object {
      "1": "4.0.1",
      "2": "5.2.0",
      "3": "missing",
      "Module": "mocha",
    },
    Object {
      "1": "11.3.0",
      "2": "missing",
      "3": "missing",
      "Module": "nyc",
    },
    Object {
      "1": "1.2.2",
      "2": "1.2.2",
      "3": "missing",
      "Module": "pre-commit",
    },
    Object {
      "1": "2.6.2",
      "2": "missing",
      "3": "missing",
      "Module": "rimraf",
    },
    Object {
      "1": "missing",
      "2": "1.8.12",
      "3": "missing",
      "Module": "bunyan",
    },
    Object {
      "1": "missing",
      "2": "4.1.2",
      "3": "missing",
      "Module": "chai",
    },
    Object {
      "1": "missing",
      "2": "3.0.1",
      "3": "missing",
      "Module": "coveralls",
    },
    Object {
      "1": "missing",
      "2": "0.0.6",
      "3": "missing",
      "Module": "eslint-config-guyellis",
    },
    Object {
      "1": "missing",
      "2": "0.4.5",
      "3": "missing",
      "Module": "istanbul",
    },
    Object {
      "1": "missing",
      "2": "0.1.6",
      "3": "missing",
      "Module": "npm-update-outdated",
    },
    Object {
      "1": "missing",
      "2": "4.0.1",
      "3": "missing",
      "Module": "rewire",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "3": "2.12.0",
      "Module": "eslint-plugin-import",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "3": "21.17.0",
      "Module": "eslint-plugin-jest",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "3": "23.1.0",
      "Module": "jest",
    },
  ],
  "peerDependencies": Array [],
}
`;

exports[`engine should get dependencies with URLs 1`] = `
Object {
  "dependencies": Array [
    Object {
      "1": "2.19.4",
      "2": "missing",
      "Module": "moment",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "2.6.1",
      "Module": "async",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "3.1.0",
      "Module": "debug",
      "some-package.json": "3.1.0",
    },
    Object {
      "1": "missing",
      "2": "4.17.10",
      "Module": "lodash",
      "some-package.json": "4.17.10",
    },
    Object {
      "1": "missing",
      "2": "4.6.5",
      "Module": "nodemailer",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "2.87.0",
      "Module": "request",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "3.3.3",
      "Module": "string",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "2.0.4",
      "Module": "sugar",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "Module": "console.table",
      "some-package.json": "0.10.0",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "Module": "minimist",
      "some-package.json": "1.2.0",
    },
  ],
  "devDependencies": Array [
    Object {
      "1": "6.26.0",
      "2": "missing",
      "Module": "babel-cli",
      "some-package.json": "missing",
    },
    Object {
      "1": "6.26.0",
      "2": "missing",
      "Module": "babel-core",
      "some-package.json": "missing",
    },
    Object {
      "1": "8.0.3",
      "2": "missing",
      "Module": "babel-eslint",
      "some-package.json": "missing",
    },
    Object {
      "1": "6.24.1",
      "2": "missing",
      "Module": "babel-preset-es2015",
      "some-package.json": "missing",
    },
    Object {
      "1": "6.24.1",
      "2": "missing",
      "Module": "babel-preset-stage-1",
      "some-package.json": "missing",
    },
    Object {
      "1": "6.26.0",
      "2": "missing",
      "Module": "babel-register",
      "some-package.json": "missing",
    },
    Object {
      "1": "4.13.1",
      "2": "4.19.1",
      "Module": "eslint",
      "some-package.json": "4.19.1",
    },
    Object {
      "1": "12.1.0",
      "2": "missing",
      "Module": "eslint-config-airbnb-base",
      "some-package.json": "12.1.0",
    },
    Object {
      "1": "1.4.0",
      "2": "missing",
      "Module": "eslint-plugin-security",
      "some-package.json": "missing",
    },
    Object {
      "1": "4.0.1",
      "2": "5.2.0",
      "Module": "mocha",
      "some-package.json": "missing",
    },
    Object {
      "1": "11.3.0",
      "2": "missing",
      "Module": "nyc",
      "some-package.json": "missing",
    },
    Object {
      "1": "1.2.2",
      "2": "1.2.2",
      "Module": "pre-commit",
      "some-package.json": "missing",
    },
    Object {
      "1": "2.6.2",
      "2": "missing",
      "Module": "rimraf",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "1.8.12",
      "Module": "bunyan",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "4.1.2",
      "Module": "chai",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "3.0.1",
      "Module": "coveralls",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "0.0.6",
      "Module": "eslint-config-guyellis",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "0.4.5",
      "Module": "istanbul",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "0.1.6",
      "Module": "npm-update-outdated",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "4.0.1",
      "Module": "rewire",
      "some-package.json": "missing",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "Module": "eslint-plugin-import",
      "some-package.json": "2.12.0",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "Module": "eslint-plugin-jest",
      "some-package.json": "21.17.0",
    },
    Object {
      "1": "missing",
      "2": "missing",
      "Module": "jest",
      "some-package.json": "23.1.0",
    },
  ],
  "peerDependencies": Array [],
}
`;

exports[`engine should get titles 1`] = `
Array [
  "1",
  "2",
  "3",
]
`;

exports[`engine should work when dependencies are identical 1`] = `
Object {
  "dependencies": Array [],
  "devDependencies": Array [],
  "peerDependencies": Array [],
}
`;
